server/migrations/20200708183925-subscription-model-upgrade.js

```diff
@@ -3,118 +3,121 @@
+            // TODO: I THINK THIS IS NOT POSSIBLE GIVEN CURRENT ISSUE IN FINDING CHAIN FROM SUBSCRIPTION
             }
-          } else if (entity === 'comment') {
-            // associate offchain_comment
-            query = `UPDATE "Subscriptions" SET offchain_comment_id=${Number(p_object_id)} WHERE id=${id};`;
-            // // associate chain or community
-            const comment = await queryInterface.sequelize.query(`SELECT * FROM "OffchainComments" WHERE id=${Number(p_object_id)};`);
-            if (comment[0].length === 0) break;
-            if (comment[0][0].chain) {
-              query += `UPDATE "Subscriptions" SET chain_id='${comment[0][0].chain}' WHERE id=${id};`;
-            } else if (comment[0][0].community) {
-              query += `UPDATE "Subscriptions" SET community_id='${comment[0][0].community}' WHERE id=${id};`;
+            break;
+          case 'new-reaction':  // object_id "comment-923" / "discussion_442" (hyphen or underscore)
+            if (entity === 'discussion') {
+              // associate offchain_thread
+              query = `UPDATE "Subscriptions" SET offchain_thread_id=${Number(p_object_id)} WHERE id=${id};`;
+              // associate chain or community
+              const thread = await queryInterface.sequelize.query(`SELECT * FROM "OffchainThreads" WHERE id=${Number(p_object_id)};`);
+              if (thread[0].length === 0) break;
+              if (thread[0][0].chain) {
+                query = `UPDATE "Subscriptions" SET chain_id='${thread[0][0].chain}' WHERE id=${id};`;
+              } else if (thread[0][0].community) {
+                query = `UPDATE "Subscriptions" SET community_id='${thread[0][0].community}' WHERE id=${id};`;
+              }
+              await queryInterface.sequelize.query(query);
+            } else if (entity === 'comment') {
+              // associate offchain_comment
+              query = `UPDATE "Subscriptions" SET offchain_comment_id=${Number(p_object_id)} WHERE id=${id};`;
+              // associate chain or community
+              const comment = await queryInterface.sequelize.query(`SELECT * FROM "OffchainComments" WHERE id=${Number(p_object_id)};`);
+              if (comment[0].length === 0) break;
+              if (comment[0][0].chain) {
+                query += `UPDATE "Subscriptions" SET chain_id='${comment[0][0].chain}' WHERE id=${id};`;
+              } else if (comment[0][0].community) {
+                query += `UPDATE "Subscriptions" SET community_id='${comment[0][0].community}' WHERE id=${id};`;
+              }
+              await queryInterface.sequelize.query(query);
+            } else {
+              // There are no new-reaction subscriptions for chain entities in our db
             }
+            break;
+          case 'chain-event': // object_id: "edgeware-reward" / "edgeware-treasury-awarded" / "edgeware-treasury-rejected"
+            query = `UPDATE "Subscriptions" SET chain_id='${entity}' WHERE id=${id};`;
```

Comment:
Since we split on hyphens as well as underscores -- `const object_id_split = object_id.split(/-|_/);` -- will this `entity` variable be potentially screwed up? It might split "edgeware-reward" into `[ "edgeware", "reward" ]`...

Reply:
yeah, so i just use the first part `entity` in the query to set the `chain_id` and then use the whole `object_id` for the `chain_event_type_id`. bc it's a `chain-event` subscription, the association is only to through those two properties which don't need the rest of the bungled up split `object_id`.

Reply:
What if the chain is `edgeware-local`?

Reply:
My suggestion here is to query the ChainEventType and use its properties, similar to how you do with threads above.

Reply:
can we just ignore `edgeware-local` as it's not in our active db? as long as the front-end and `/createSubscription` handle future case of hyphenated chains entities?

Reply:
That's true -- fair enough!